src/main/resources/META-INF/rewrite/was-to-liberty.yml

```
@@ -0,0 +1,41 @@
+#
+# Copyright 2023 the original author or authors.
+# <p>
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# <p>
+# https://www.apache.org/licenses/LICENSE-2.0
+# <p>
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+########################################################################################################################
+# WebSphere traditional to Liberty
+---
+type: specs.openrewrite.org/v1beta/recipe
+name: org.openrewrite.java.liberty
+displayName: Migrate from WebSphere traditional to Liberty
+description: >
+  Use this category of rules to identify code changes needed when migrating
+  from WebSphere Application Server traditional to Liberty.
+recipeList:
+ - org.openrewrite.java.liberty.WebSphereUnavailableSSOCookieMethod
+ - org.openrewrite.java.liberty.WebSphereUnavailableSSOTokenMethod
+---
+type: specs.openrewrite.org/v1beta/recipe
+name: org.openrewrite.java.liberty.WebSphereUnavailableSSOCookieMethod
+recipeList:
+  - org.openrewrite.java.liberty.ChangeMethodObjectInvocation
+---
+type: specs.openrewrite.org/v1beta/recipe
+name: org.openrewrite.java.liberty.WebSphereUnavailableSSOTokenMethod
+recipeList:
+  - org.openrewrite.java.liberty.ChangeMethodInvocation:
+      methodPattern: com.ibm.websphere.security.WSSecurityHelper getLTPACookieFromSSOToken()
+      newMethodPattern: com.ibm.websphere.security.web.WebSecurityHelper getSSOCookieFromSSOToken()
+      performStaticCall: true
```

I have a strong feeling that for this use case we can merely use the existing
https://docs.openrewrite.org/recipes/java/changemethodname
Would like to make sure before we add `ChangeMethodInvocation` here, even if we add that later in #2 , just to keep complexity down where we can. Once the tests are running we should be good to try this out.

---

The ChangeMethodName recipe based on the documentation and usage in various repositories only changes the method name. It doesn't change the type of the variable invoking the method or the arguments. In the recipe repo, there exist recipes that will change the variable type, re-order the arguments in the method or delete arguments. But nothing currently exists to add a new argument to a method. For our use case, it was easier to provide a recipe to do all of this. If you are aware of a recipe combination we can use to provide the functionality, we would be more than happy to use that instead.

---

Ah I'd missed the addition of `.web.`. Then you can just use two recipes in Yaml back-to-back.
```suggestion
  - org.openrewrite.java.ChangeMethodName:
      methodPattern: com.ibm.websphere.security.WSSecurityHelper getLTPACookieFromSSOToken()
      newMethodName: getSSOCookieFromSSOToken
  - org.openrewrite.java.ChangeType:
      oldFullyQualifiedTypeName: com.ibm.websphere.security.WSSecurityHelper
      newFullyQualifiedTypeName: com.ibm.websphere.security.web.WSSecurityHelper
```
You can compose such recipes using our builder: https://app.moderne.io/recipes/builder
That typically can do a lot of these combined changes easily.

Sometimes you have to swap the order; I haven't tried the above yet, but if it fails you might want to swap the recipe order and add `.web.` into the `methodPattern` argument.

---

sure Tim will try it out..guess let Chuka know in slack as well.